### PR TITLE
app/vmselect/graphite: enforce search.maxQueryLen for Graphite queries

### PR DESCRIPTION
**Pull request by camathieu**
Description:

This PR ensures that the -search.maxQueryLen flag applies to Graphite queries, matching the behavior already present for Prometheus queries. Previously, Graphite queries could bypass this limit, creating an inconsistency and a potential vector for resource exhaustion.

Key changes:

Added getMaxQueryLen() to access the global query length limit.
Enforced query length validation in execExpr() for Graphite queries.
Added comprehensive tests for the new validation logic and edge cases.
Error messages are consistent with Prometheus query validation.
The default limit is 16KB (configurable via -search.maxQueryLen). Setting the limit to 0 disables validation.
This change closes the gap where Graphite queries could exceed configured length limits, providing consistent protection against excessively long queries across both query APIs.

Related to https://github.com/VictoriaMetrics/VictoriaMetrics/pull/9534

### CODE REVIEW

File: app/vmselect/graphite/eval.go [hunk 1]
```diff
@@ -21,6 +22,18 @@ var maxGraphiteSeries = flag.Int("search.maxGraphiteSeries", 300e3, "The maximum
 var maxGraphitePathExpressionLen = flag.Int("search.maxGraphitePathExpressionLen", 1024, "The maximum length of pathExpression field in Graphite series. "+
 	"Longer expressions are truncated to prevent memory exhaustion on complex nested queries. Set to 0 to disable truncation.")
 
+// getMaxQueryLen returns the current value of the search.maxQueryLen flag.
+// This flag is defined in the prometheus package, but we need to access it for Graphite queries too.
+func getMaxQueryLen() int {
+	// Get the flag value directly from the flag package
+	if f := flag.Lookup("search.maxQueryLen"); f != nil {
```
Comment (f41gh7) [inline app/vmselect/graphite/eval.go]: I think, it's better to move flag `search.maxQueryLen` into `vmselect/searchutil` package and expose it's value with `GetMaxQueryLen` function.

**Comment by f41gh7** [comment]
Could you please also mention changes at `changlog`? It's changes behavior of graphite API and it should be mentioned.

**Comment by camathieu** [comment]
PR updated

**Comment by f41gh7** [comment]
Thanks for contribution!

**Comment by f41gh7** [comment]
The feature was included into [v1.125.0](https://docs.victoriametrics.com/changelog/#v11250) release.
